ext/dom/document.c

```diff
@@ -1620,55 +1621,45 @@ PHP_FUNCTION(dom_document_savexml)
 	if (zend_parse_method_parameters(ZEND_NUM_ARGS(), getThis(), "O|O!l", &id, dom_document_class_entry, &nodep, dom_node_class_entry, &options) == FAILURE) {
 		return;
 	}
+	options = options | XML_SAVE_AS_XML;
 
 	DOM_GET_OBJ(docp, id, xmlDocPtr, intern);
 
 	doc_props = dom_get_doc_props(intern->document);
 	format = doc_props->formatoutput;
 
+	buf = xmlBufferCreate();
+	if (!buf) {
+		php_error_docref(NULL, E_WARNING, "Could not fetch buffer");
+		RETURN_FALSE;
+	}
+	xmlSaveCtxtPtr xscp = xmlSaveToBuffer(buf, docp->encoding, options);
```

Comment:
C89 doesn't allow mixing declarations and statements, so please move the declaration up to the start of the block.